qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -411,32 +411,42 @@ def supports_aux_operators(cls) -> bool:
     def _eval_aux_ops(
         self,
         parameters: np.ndarray,
-        aux_operators: List[OperatorBase],
+        aux_operators: ListOrDict[OperatorBase],
         expectation: ExpectationBase,
         threshold: float = 1e-12,
-    ) -> np.ndarray:
+    ) -> ListOrDict[complex]:
         # Create new CircuitSampler to avoid breaking existing one's caches.
         sampler = CircuitSampler(self.quantum_instance)
 
-        aux_op_meas = expectation.convert(StateFn(ListOp(aux_operators), is_measurement=True))
+        if isinstance(aux_operators, dict):
+            list_op = ListOp(list(aux_operators.values()))
```

Comment:
Do we need to use an `OrderedDict` to ensure that this does not break?

Reply:
Starting from 3.6 a standard Python `dict` maintains the insertion order so I thought it would not be necessary to use an `OrderedDict`
https://stackoverflow.com/a/39537308/13115502

Reply:
Thanks, that's very good to know! :+1: